### PR TITLE
[HW23] Mongo DB

### PR DESCRIPTION
**Pull request by gamzinova**
Hi,
Please find my homework 24.
I was trying to separate all the connections in the pytest fixture and add JSON configuration file and its parser, but not succeed.
Is the reason for my failure that I am not creating any pytest test file, and fixtures can be used only there?
Anyway, you have asked for the homework the main.py file with printing the results.
Thanks in advance for the explanation.
Best,
Anna

### CODE REVIEW

File: Anna_Gamzinova/MongoDB/objects/base_db.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import pymongo
+
+
+class BaseDB:
+    def __init__(self, database, collection):
+        self.connection = pymongo.MongoClient("mongodb://127.0.0.1:27017/")
+        self.database = self.connection[database]
+        self.collection = self.database[collection]
+
+    def insert_value(self, new_name, new_color, new_occupation):
```
Comment (bohdandovbsyh) [inline Anna_Gamzinova/MongoDB/objects/base_db.py:10]: it is not a base function. 
What if I want to insert value for other instance which do not have such fields?

File: Anna_Gamzinova/MongoDB/objects/base_db.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    def insert_values(self, my_tuples: tuple):
+        """
+        list should be filled with dicts, with such format {'name': new_name, 'color': new_color, 'occupation':
+        new_occupation}
+        """
+        return self.collection.insert_many(
+            [{'name': insert_i[0], 'color': insert_i[1], 'occupation': insert_i[2]} for insert_i in my_tuples])
+
+    def find_value(self):
+        return self.collection.find_one()
```
Comment (bohdandovbsyh) [inline Anna_Gamzinova/MongoDB/objects/base_db.py:25]: find_one works with filter. You can pass dict to it to find specific value

File: Anna_Gamzinova/MongoDB/objects/base_db.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    def find_all_values(self):
+        return [x for x in self.collection.find()]
+
+    def delete_item_by_name(self, del_name: str):
```
Comment (bohdandovbsyh) [inline Anna_Gamzinova/MongoDB/objects/base_db.py:30]: what if the item does not have a name? 
More general is deleted by id.

File: Anna_Gamzinova/MongoDB/objects/base_db.py (hunk 1 above)
Comment (gamzinova) [reply Anna_Gamzinova/MongoDB/objects/base_db.py]: @bohdandovbsyh removed to cars

File: Anna_Gamzinova/MongoDB/objects/base_db.py (hunk 2 above)
Comment (gamzinova) [reply Anna_Gamzinova/MongoDB/objects/base_db.py]: fixed

File: Anna_Gamzinova/MongoDB/objects/base_db.py (hunk 3 above)
Comment (gamzinova) [reply Anna_Gamzinova/MongoDB/objects/base_db.py]: fixed

**Pull request merged**